src/Ryujinx.HLE/HOS/Services/Ldn/UserServiceCreator/IUserLocalCommunicationService.cs

```diff
@@ -1,88 +1,1099 @@
+                            disconnectReason = DisconnectReason.DestroyedBySystem;
+                        }
+                        else
+                        {
+                            disconnectReason = DisconnectReason.DestroyedByUser;
+                        }
+
+                        DestroyNetworkImpl(disconnectReason);
+
+                        break;
+                    }
+                case NetworkState.Station:
+                    {
+                        CloseStation();
+
+                        break;
+                    }
+                case NetworkState.StationConnected:
+                    {
+                        if (isCausedBySystem)
+                        {
+                            disconnectReason = DisconnectReason.DisconnectedBySystem;
+                        }
+                        else
+                        {
+                            disconnectReason = DisconnectReason.DisconnectedByUser;
+                        }
+
+                        DisconnectImpl(disconnectReason);
+
+                        break;
+                    }
+            }
+
+            SetState(NetworkState.None);
+
+            NetworkClient?.DisconnectAndStop();
+            NetworkClient = null;
+
+            return ResultCode.Success;
         }
 
         [CommandCmif(402)] // 7.0.0+
-        // Initialize(u64 ip_addresses, u64, pid)
+        // Initialize(u64 ip_addresses, pid)
         public ResultCode Initialize(ServiceCtx context)
         {
-            // TODO(Ac_K): Determine what addresses are.
-            IPAddress unknownAddress1 = new(context.RequestData.ReadUInt32());
-            IPAddress unknownAddress2 = new(context.RequestData.ReadUInt32());
+            _ = new IPAddress(context.RequestData.ReadUInt32());
+            _ = new IPAddress(context.RequestData.ReadUInt32());
+
+            // NOTE: It seems the guest can get ip_address and subnet_mask from nifm service and pass it through the initialize.
+            //       This call twice InitializeImpl(): A first time with NIFM_REQUEST_ID, if its failed a second time with nifm_request_id = 1.
+            //       Since we proxy connections, we don't needs to get the ip_address, subnet_mask and nifm_request_id.
+
+            return InitializeImpl(context, context.Process.Pid, NifmRequestID);
+        }
+
+        public ResultCode InitializeImpl(ServiceCtx context, ulong pid, int nifmRequestId)
+        {
+            ResultCode resultCode = ResultCode.InvalidArgument;
+
+            if (nifmRequestId <= 255)
+            {
+                if (_state != NetworkState.Initialized)
+                {
+                    // NOTE: Service calls nn::ldn::detail::NetworkInterfaceManager::NetworkInterfaceMonitor::Initialize() with nifmRequestId as argument.
+                    //       Then it stores the result code of it in a global field. Since we use our own implementation, we can just check the connection
```

Comment:
```suggestion
                    //       then it stores the result code of it in a global field. Since we use our own implementation, we can just check the connection
```
Maybe it should say "global variable" rather than "global field"?